control/Director.php

```diff
@@ -422,13 +422,23 @@ public static function set_current_page($page) {
 	 */
 	public static function absoluteURL($url, $relativeToSiteBase = false) {
 		if(!isset($_SERVER['REQUEST_URI'])) return false;
-		
+
+		//a url of . or ./ is the same as an empty url
+/*
+		if ($url == '.' || $url == './') {
+			$url = '';
+		}
+*/		
 		if(strpos($url,'/') === false && !$relativeToSiteBase) {
-			$url = dirname($_SERVER['REQUEST_URI'] . 'x') . '/' . $url;
+			//if there's no URL we want to force a trailing slash on the link
+			if (!$url) {
```

Comment:
If you add the '/' to the last parameter of join_links it ensures the url ends with a slash, but avoids double slashes.

`$url = Controller::join_links(dirname($_SERVER['REQUEST_URI'] . 'x'), $url, '/');`